model/task/task.go

```diff
@@ -248,6 +248,13 @@ type Task struct {
 
 	CanSync       bool             `bson:"can_sync" json:"can_sync"`
 	SyncAtEndOpts SyncAtEndOptions `bson:"sync_at_end_opts,omitempty" json:"sync_at_end_opts,omitempty"`
+
+	// IsEssentialToFinish indicates that this task must finish in order for
+	// the build and version to be considered complete. For example, tasks
+	// selected by the GitHub PR alias must succeed for the GitHub PR requester
+	// before its build or version can be considered complete, but tasks
+	// manually scheduled by the user afterwards are not required.
```

Comment:
Does this mean that the version status will also be updated only off of this, or overall will the patch status wait until this is done? 

I'd prefer if we had this wording / usage be more similar to IsGithubCheck, especially since we're only using it for PR patches / statuses.

Reply:
All the higher-level statuses (build, patch, version) will remain in "running" until the required tasks are all done.

I was thinking this is a pretty general/extensible feature since it's not doing any GitHub-specific actions like `IsGithubCheck` is, but if we think it's only useful for GitHub PR check marks and has no other use cases, I can rename it.

Reply:
Oh I see -- I didn't read the PR closely and misunderstood that we weren't special casing this for Github statuses.

Can we reconsider that? I think it's pretty important to users that Evergreen's statuses reflect what is actually happening in Evergreen. I also think that's reasonable -- the thing we want to do here is make sure that the Github check we're sending accurately reflects what they've asked us to check for Github, which can be separate from what we're doing in Evergreen overall. It seems like we could compute a status here, or check if all tasks essential to finish have been completed at the time of sending the status in triggers (and perhaps we could send a special notification in refresh like "not all required checks are scheduled" as a red check or pending status if we're concerned about people spamming it).

Reply:
I feel like this would be doable but I would also think that the patch being in an unfinished state (running) _is_ the patch's true state. Sending special alternative statuses for a specific condition seems like it would add more overhead to think about than I think is needed for something which is preventing unusual/atypical user behavior (at least, I can't think of a workflow where a user would want to intentionally unschedule required PR tasks). Since the build/version updating logic is pretty far from the GitHub status job, we would have to either encode this as another field/status in the build/version to represent "done but we can't tell GitHub that you're all good" (which I'd like to avoid) or dynamically re-check all the GitHub essential tasks in the trigger notification logic (which also seems very special case).

If it turns out people have a lot of questions about this and don't see the note in the docs, I think we can revisit making it more ergonomic later and consider sending a special notification just for this state. We could do something like add a display status in Spruce representing tasks that are essential but haven't run yet (like add an asterisk to them or something).

Reply:
Alright, I thought it over, and I'm gonna leave the build/patch in the running state but see if I can add some way of notifying the user of which/how many essential tasks still need to run.

Reply:
Lmk when I should look at this PR again @Kimchelly !

Reply:
It should be ready now.

Reply:
Sorry I missed this -- will look first thing tomorrow!